src/test/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/jpa/repository/PrisonSignedOperationalCapacityRepositoryTest.kt

```diff
@@ -0,0 +1,43 @@
+package uk.gov.justice.digital.hmpps.locationsinsideprison.jpa.repository
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
+import org.springframework.test.context.jdbc.Sql
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.locationsinsideprison.integration.TestBase
+import java.time.LocalDateTime
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@Transactional
+class PrisonSignedOperationalCapacityRepositoryTest : TestBase() {
+
+  @Autowired
+  lateinit var repository: PrisonSignedOperationCapacityRepository
+
+  @AfterEach
+  fun cleanUp() {
+    repository.deleteAll()
+  }
+
+  @Test
+  fun `Return null when capacity not defined for prison id`() {
+    var oc = repository.findOneByPrisonId("MDI")
+    assertThat(oc).isNull()
+  }
+
+  @Test
+  @Sql("classpath:repository/insert-prison-signed-operation-capacity.sql")
+  fun `Return result when capacity defined for prison id`() {
+    var oc = repository.findOneByPrisonId("MDI")
+    assertThat(oc?.id).isNotNull()
```

Comment:
could throw before this - to stop the additional ? in each line

Reply:
I don't like the idea to throw the exception to satisfy business logic. The throwing the exception is very slow comparing to check to null. So Im more happy to return null when record not found as we need this in the process of the creation SignedOperationCapacityService line 34.